### PR TITLE
Convert to onEvent

### PR DESCRIPTION
**Pull request by tiina303**
We're moving away from exportEvents and a custom buffer to batch exports or onEvent apps.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -89,20 +87,6 @@ async function exportEvents(events, { config, global }) {
             }
         }
     }
-
-}
-
-async function fetchWithRetry(url, options = {}, method = 'GET', isRetry = false) {
```
Comment (tiina303) [inline index.js:95]: the retry part was never used, removing it as it shouldn't belong in the apps code in the first place.

Comment (bretthoerner) [reply index.js:95]: It was used though, right? It looks like it would retry once.

Why doesn't it belong in apps code? Do we handle retries for the apps?

Comment (tiina303) [reply index.js:95]: it wasn't isRetry is always false

Comment (tiina303) [reply index.js:95]: yup exactly we're handling retries internally ...

Comment (bretthoerner) [reply index.js:95]: > it wasn't isRetry is always false

It starts as `false` and then if there's an error it does 1 retry with `isRetry = true`, and if it errors again it gives up. So I just wanted to call out that this is a functionality change technically.

**Review by bretthoerner — APPROVED** [review]
Makes sense to me as long as you're OK removing the retry.

File: index.js (hunk 1 above)
Comment (tiina303) [reply index.js:95]: oh yes ... I see it now 🤦‍♀️ , but if we want retries we should just throw retryError on the specific errors that should be Retriable not all errors

**Pull request merged**